frontend/src/stores/feature-flag.ts

```diff
@@ -122,34 +121,13 @@ export const useFeatureFlagStore = defineStore(FEATURE_FLAG, {
      *
      * @param cookies - mapping of feature flags and their preferred states
      */
-    initFromCookies(cookies: Record<string, FeatureState>) {
+    initFromCookies(cookies: Record<string, FeatureState>[]) {
       Object.entries(this.flags).forEach(([name, flag]) => {
-        if (getFlagStatus(flag) === SWITCHABLE && flag.storage === COOKIE)
-          Vue.set(flag, "preferredState", cookies[name])
-      })
-    },
-    /**
-     * Write the current state of the switchable flags to the cookie.
-     *
-     * @param cookies - the Nuxt cookies module
-     */
-    writeToCookies(cookies: Context["$cookies"]) {
-      cookies.set("features", this.flagStateMap(COOKIE))
-    },
-    /**
-     * Initialize the state of the switchable flags from the session storage.
-     * `Vue.set` is used to ensure reactivity is maintained.
-     */
-    initFromSession() {
-      if (typeof window === "undefined") return
-      const features = useStorage<Record<string, FeatureState>>(
-        "features",
-        {},
-        sessionStorage
-      )
-      Object.entries(this.flags).forEach(([name, flag]) => {
-        if (getFlagStatus(flag) === SWITCHABLE && flag.storage === SESSION) {
-          Vue.set(flag, "preferredState", features.value[name])
+        if (getFlagStatus(flag) === SWITCHABLE) {
+          const cookie = cookies.find((cookie) =>
```

Comment:
Can the finding of the cookie value be simplified here? I don't fully understand what's going on.